apiserver/facades/client/cloud/cloud_test.go

```diff
@@ -1312,17 +2097,35 @@ func (s *cloudSuite) TestModifyCloudUpdateAccess(c *gc.C) {
 		},
 	})
 	c.Assert(err, jc.ErrorIsNil)
-	s.backend.CheckCallNames(c, "Cloud", "ControllerTag", "CreateCloudAccess", "GetCloudAccess", "UpdateCloudAccess")
-	s.backend.CheckCall(c, 2, "CreateCloudAccess", "fluffy", names.NewUserTag("fred"), permission.AdminAccess)
-	s.backend.CheckCall(c, 4, "UpdateCloudAccess", "fluffy", names.NewUserTag("fred"), permission.AdminAccess)
 	c.Assert(results.Results, gc.DeepEquals, []params.ErrorResult{
 		{},
 	})
 }
 
-func (s *cloudSuite) TestModifyCloudAlreadyHasAccess(c *gc.C) {
-	s.backend.cloudAccess = permission.AdminAccess
-	results, err := s.api.ModifyCloudAccess(params.ModifyCloudAccessRequest{
+func (s *cloudSuiteGomocks) TestModifyCloudAlreadyHasAccess(c *gc.C) {
+	bruceTag := names.NewUserTag("bruce")
+	defer s.setup(c, bruceTag).Finish()
+
+	cloud := cloud.Cloud{
+		Name:      "fluffy",
+		Type:      "fluffy",
+		AuthTypes: []cloud.AuthType{cloud.EmptyAuthType, cloud.UserPassAuthType},
+		Regions:   []cloud.Region{{Name: "nether", Endpoint: "endpoint"}},
+	}
+
+	fredTag := names.NewUserTag("fred")
+
+	backend := s.backend.EXPECT()
+	backend.ControllerTag().Return(coretesting.ControllerTag)
+	backend.Cloud("fluffy").Return(cloud, nil)
+	backend.CreateCloudAccess("fluffy", fredTag,
+		permission.AdminAccess).Return(errors.New("user already has \"admin\" access or greater"))
```

Comment:
Where is this return error set up in the original test?

Reply:
https://github.com/juju/juju/blob/29240734bf5a191fc95ba19cb715470785c111af/apiserver/facades/client/cloud/cloud_test.go#L1340